### PR TITLE
Adopt format of Linux MAINTAINERS file

### PR DESCRIPTION
**Pull request by jforissier**
Our current MAINTAINERS.md file is quite impractical for two common use
cases:

1. Finding the platform maintainer(s) given a file path in the OP-TEE
OS source tree;
2. Contacting maintainers via GitHub (such as mentioning them in pull
requests).

To fix this, adopt the Linux format and extend the e-mail address field
to also contain an optional GitHub ID, such as:

  John Doe <john.doe@example.com> [@johndoe]

Since the file is not in Markdown format anymore, drop the .md
extension as well.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jforissier** [comment]
@nodeax @kevin-peng-hao @b49020 @pangupta @yanyan-wrs @MrVan @lorc @HarinarayanaBhatta @glneo @sorenb-xlnx 

As you can see, I have added your GitHub IDs to the MAINTAINERS file. Ideally I'd like an `Acked-by:` from each and every one of you (or if for any reason you don't want to associate your GitHub ID with your email in this file, just let me know). Thanks!

**Comment by lorc** [comment]
Hi @jforissier,

I generally support this change, but maybe it is worth to list not only platform maintainers, but whole  OP-TEE mainteiners as well?

At least - add  `THE REST` entry.

File: MAINTAINERS [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+M:	Linaro <op-tee@linaro.org>
+S:	Maintained
+F:	core/arch/arm/plat-stm/
+
+Texas Instruments DRA7xx
+M:	Harinarayan Bhatta <harinarayan@ti.com> [@HarinarayanaBhatta]
+S:	Maintained
+F:	core/arch/arm/plat-ti/
+
+Texas Instruments AM43xx
```
Comment (glneo) [inline MAINTAINERS]: Harinarayan is no longer a maintainer for these platforms, you can assign all plat-ti to me here.

Texas Instruments AM43xx, AM57xx, DRA7xx

Other than that,

Acked-by: Andrew F. Davis <afd@ti.com>

**Comment by jforissier** [comment]
@lorc yes good idea, will add an entry.
@glneo will fix.

**Comment by jforissier** [comment]
@jenswi-linaro @d3zd3z @etienne-lms @igoropaniuk @jbech-linaro @vchong 
FYI, I have added your name/email/GitHub ID to "THE REST".

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <vlad.babchuk@epam.com>`

File: MAINTAINERS [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+F:	core/arch/arm/plat-ti/
+
+Xilinx Zynq 7000 ZC702 Board
+R:	Yan Yan <yan.yan@windriver.com> [@yanyan-wrs]
+R:	Feng Yu <Yu.Feng@windriver.com>
+S:	Maintained
+F:	core/arch/arm/plat-zynq7k/
+
+Xilinx Zynq UltraScale+ MPSOC
+R:	Sören Brinkmann <soren.brinkmann@xilinx.com> [@sorenb-xlnx]
```
Comment (jenswi-linaro) [inline MAINTAINERS]: According to Linked it looks like Sören might not do this any more.

Comment (jbech-linaro) [reply MAINTAINERS]: Aha, that might explain why we haven't heard from him in a while. We can ask him if Apple have any device he wants to put here ;-)

Comment (jforissier) [reply MAINTAINERS]: ;-)
OK I'll mark the platform as Orphan, then.

**Comment by etienne-lms** [comment]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`

**Comment by nodeax** [comment]
`Acked-by: Akshay Bhat <akshay.bhat@timesys.com>`

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

File: MAINTAINERS [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+S:	Maintained
+F:	core/arch/arm/plat-hikey/
+
+HiSilicon Poplar (Hi3798C V200)
+R:	Linaro <op-tee@linaro.org>
+S:	Maintained
+F:	core/arch/arm/plat-poplar/
+
+Marvell Armada 70x0, Armada 80x0, Armada 3700
+R:	Kevin Peng <kevinp@marvell.com> [@kevin-peng]
```
Comment (kevin-peng-hao) [inline MAINTAINERS]: Should be @kevin-peng-hao

**Comment by MrVan** [comment]
`Acked-by: Peng Fan <peng.fan@nxp.com>`

File: MAINTAINERS (hunk 3 above)
Comment (jforissier) [reply MAINTAINERS]: Fixed, thanks.

**Comment by kevin-peng-hao** [comment]
`Acked-by: Kevin Peng <kevinp@marvell.com>`

**Comment by b49020** [comment]
Acked-by: Sumit Garg <sumit.garg@nxp.com>

**Comment by jforissier** [comment]
Ping: @d3zd3z @igoropaniuk @jbech-linaro @vchong @pangupta @yanyan-wrs @glneo
Can you please give an `Acked-by:`? Otherwise I will remove the changes that are related to you (i.e., not mention your GitHub ID or not add you to "THE REST").
Thanks!

**Comment by jbech-linaro** [comment]
Oh, I thought you were just asking for Acked-by from external people, sorry for misunderstanding that.
`Acked-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by vchong** [comment]
Me too.
`Acked-by: Victor Chong <victor.chong@linaro.org>`

**Comment by igoropaniuk** [comment]
`Acked-by: Igor Opaniuk <igor.opaniuk@linaro.org>`

**Pull request merged**